src/pages/registration/contributors_tab/components/AddAffiliationModal.tsx

```diff
@@ -0,0 +1,119 @@
+  setAffiliationToVerify: (val: string) => void;
+  toggleAffiliationModal: () => void;
+  authorName: string;
+  affiliationToVerify: string;
+  affiliations?: Affiliation[];
+  baseFieldName: string;
+}
+
+export const AddAffiliationModal = ({
+  openAffiliationModal,
+  setAffiliationToVerify,
+  toggleAffiliationModal,
+  affiliationToVerify,
+  authorName,
+  affiliations = [],
+  baseFieldName,
+}: AddAffiliationModalProps) => {
+  const { t } = useTranslation();
+  const { setFieldValue, values } = useFormikContext<Registration>();
+  const dispatch = useDispatch();
+
+  const addAffiliation = (newAffiliationId: string) => {
+    if (!newAffiliationId) {
+      return;
+    }
+
+    // Avoid adding same unit twice
+    if (
+      affiliations.some((affiliation) => affiliation.type === 'Organization' && affiliation.id === newAffiliationId)
+    ) {
+      dispatch(setNotification({ message: t('registration.contributors.add_duplicate_affiliation'), variant: 'info' }));
+      return;
+    }
+
+    const addedAffiliation: Affiliation = {
+      type: 'Organization',
+      id: newAffiliationId,
+    };
+
+    let updatedAffiliations = [...affiliations]; // Must spread affiliations in order to keep changes when switching tab
+
+    if (affiliationToVerify) {
+      // Verify affiliation
+      const affiliationIndex = updatedAffiliations.findIndex(
+        (affiliation) => affiliation.type === 'UnconfirmedOrganization' && affiliation.name === affiliationToVerify
+      );
+      updatedAffiliations[affiliationIndex] = addedAffiliation;
+    } else {
+      // Add new affiliation
+      if (updatedAffiliations) {
+        updatedAffiliations.push(addedAffiliation);
+      } else {
+        updatedAffiliations = [addedAffiliation];
+      }
+    }
+
+    setFieldValue(`${baseFieldName}.${SpecificContributorFieldNames.Affiliations}`, updatedAffiliations);
+    toggleAffiliationModal();
+  };
+
+  return (
+    <Modal
+      open={openAffiliationModal}
+      onClose={() => {
+        setAffiliationToVerify('');
+        toggleAffiliationModal();
+      }}
+      maxWidth="md"
+      fullWidth={true}
+      headingText={t('registration.contributors.add_new_affiliation')}
+      dataTestId="affiliation-modal">
+      <Trans
+        i18nKey="registration.contributors.add_new_affiliation_helper_text"
+        components={[<Typography paragraph />]}
+      />
+      <Box
+        sx={{
+          bgcolor: 'secondary.main',
+          borderRadius: '0.25rem',
+          padding: '0.5rem 0.75rem',
+          marginBottom: '2rem',
+        }}>
+        <Typography>
+          {t('common.contributor')}: <b>{authorName}</b>
+        </Typography>
+      </Box>
+      {affiliationToVerify && (
+        <Typography paragraph>
+          {t('registration.contributors.prefilled_affiliation')}: <b>{affiliationToVerify}</b>
+        </Typography>
+      )}
+      <Typography variant="h3" component="h2" sx={{ marginBottom: '1rem', fontWeight: 'normal' }}>
```

Comment:
Naturlig at denne er på samme nivå som "Velg enhet", og ikke selve modal-headeren(?)
```suggestion
      <Typography variant="h3" sx={{ marginBottom: '1rem', fontWeight: 'normal' }}>
```

Reply:
Ja den må også være h3